### DISCUSSION TITLE
Handling of empty arrays

### DISCUSSION THREAD

**Discussion by JDziurlaj** [frictionlessdata/frictionlessdata.io#634] [Q&A] [answered]
How should a `tableschema` aware tool interpret the existence of an empty array (`[]`)?

Should it be considered:

- logically null
- logically not-null (i.e. it meets the `required` constraint)
- something else

If it is logically null, how should `[ null ]` be handled?

**Comment by rufuspollock** [comment]
What's the exact situation - what is the type of the field where this showing up and what constraints are there?

**Comment by roll** [comment, marked as answer]
Hi @JDziurlaj,

The empty array is not null unless its string representation is IN the missing values (for textual source like csv/etc) - https://specs.frictionlessdata.io/table-schema/#missing-values

So it will pass the `required` constraint but we also have `minLength` constraint to handle cases like this.
